gems/aws-sdk-core/lib/aws-sdk-core/plugins/client_metrics_plugin.rb

```diff
@@ -0,0 +1,191 @@
+require 'date'
+
+module Aws
+  module Plugins
+    class ClientMetricsPlugin < Seahorse::Client::Plugin
+
+      option(:client_side_monitoring,
+        default: true,
+        docstring: <<-DOCS) do |cfg|
+When `true`, client-side metrics will be collected for all API requests from
+this client.
+      DOCS
+        resolve_client_side_monitoring(cfg)
+      end
+
+      option(:client_side_monitoring_port,
+        default: nil,
+        docstring: <<-DOCS) do |cfg|
+Required for publishing client metrics. The port that the client side monitoring
+agent is running on, where client metrics will be published via UDP.
+      DOCS
+        resolve_client_side_monitoring_port(cfg)
+      end
+
+      option(:client_side_monitoring_publisher,
+        default: ClientSideMonitoring::Publisher.new,
+        docstring: <<-DOCS)
+Allows you to provide a custom client-side monitoring publisher class. By default,
```

Comment:
So by default `client_side_monitoring_publisher` enabled but without a port unless `client_side_monitoring_port `/ENV is specified? Then actually it's not working by pure default?